### PR TITLE
fix: Stability improvements [BREAKING CHANGE]

### PR DESCRIPTION
**Pull request by LeagueOfPoro**
- During each heartbeat, the bot decodes access_token and retrieves its expiration time. If the time left is under 10 minutes (total validity is 60 min) it refreshes the session.
- Added 30s cooldown between thread restarts

BREAKING CHANGE: New dependency `pyjwt` to decode the tokens. Run `pipenv install` to install the new dependency.

### CODE REVIEW

File: Browser.py [hunk 1]
```diff
@@ -196,10 +186,21 @@ def checkNewDrops(self, lastCheckTime):
...
+    def __needSessionRefresh(self) -> bool:
+        if "access_token" in self.client.cookies.get_dict():
+            res = jwt.decode(self.client.cookies.get_dict()["access_token"], options={"verify_signature": False})
+            timeLeft = res['exp'] - int(time())
+            self.log.debug(f" s until session expires.")
+            if timeLeft < 600:
+                return True
+            return False
+        else:
+            raise NoAccessTokenException()
```
Comment (DjordyKoert) [inline Browser.py:202]: This can help clarify your code by using an early return.

```suggestion
        if "access_token" not in self.client.cookies.get_dict():
            raise NoAccessTokenException()

        res = jwt.decode(self.client.cookies.get_dict()["access_token"], options={"verify_signature": False})
        timeLeft = res['exp'] - int(time())
        self.log.debug(f" s until session expires.")
        if timeLeft < 600:
            return True
        return False
```

File: FarmThread.py [hunk 2]
```diff
@@ -60,8 +56,11 @@ def run(self):
                     sleep(Browser.STREAM_WATCH_INTERVAL)
             else:
                 self.log.error(f"Login for {self.account} FAILED!")
-                self.stats.updateStatus(self.account, "[red]LOGIN FAILED")
                 self.stats.addLoginFailed(self.account)
+                if self.stats.getFailedLogins(self.account) < 3:
+                    self.stats.updateStatus(self.account, "[red]LOGIN FAILED - WILL RETRY SOON")
```
Comment (DjordyKoert) [inline FarmThread.py:61]: These statuses could be placed inside an Enum. https://docs.python.org/3/library/enum.html
This can help re-use these statuses (Maybe do this in a separate PR?)

Comment (LeagueOfPoro) [reply FarmThread.py:61]: I will do it in separate PR

File: main.py [hunk 3]
```diff
@@ -53,6 +54,9 @@
         for account in config.accounts:
             if account not in farmThreads:
                 if stats.getFailedLogins(account) < 3:
+                    if stats.getFailedLogins(account) > 0:
+                        log.debug("Sleeping {account} before retrying login.")
+                        sleep(30)
```
Comment (Jang0o184) [inline main.py:64]: Maybe could be a good Idea to have a long term solution after this fails
20-30 minutes or maybe even longer before trying to login after this
Then it would be able to handle even longer crashes
Maybe (but I might be too stupid) adding some kind of random delay +-10 minutes to ensure that not every crashed bot tries to login after exactly 30 minutes for example.
